### ISSUE TITLE
BUG: Make order of `"states"` in model specification more flexible

### ISSUE THREAD

**Issue by ChristianZimpelmann** [OpenSourceEconomics/lcm#40]
If in the model-specification dictionary a continuous state variable is followed by a discrete state variable (see below), the following error is shown:

`ValueError: Interpolation axes need to be the last entries in axis_order.`

#### Code Sample, a copy-pastable example

```python
    "states": {
        "wealth": {
            "grid_type": "linspace",
            "start": 100,
            "stop": 10_000,
            "n_points": 2000,
        },
        "age_youngest_child": {"options": list(range(19))},
    },
```

If the order is reversed, it works fine.

#### Expected Output

- At some point lcm should probably reorder the state variable definitions internally such that no error is thrown
- Alternatively, at least the error message should be improved

Bug does not have a high priority (should we add a label to indicate that?)

**Comment by janosg** [comment]
Good that you made a bug report. This should have worked already and should be easy to fix if not.

**Comment by timmens** [comment]
Solved in #42
